packages/react-components/react-migration-v0-v9/src/components/List/ListItem/useListItem.tsx

```diff
@@ -0,0 +1,127 @@
+import { useARIAButtonProps } from '@fluentui/react-aria';
+import { useFocusableGroup } from '@fluentui/react-tabster';
+import { getIntrinsicElementProps, slot, useEventCallback, useId } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { useListContext_unstable } from '../List/listContext';
+import type { ListItemProps, ListItemState } from './ListItem.types';
+
+const DEFAULT_ROOT_EL_TYPE = 'li';
+
+function validateProperElementTypes(parentRenderedAs?: 'div' | 'ul' | 'ol', renderedAs?: 'div' | 'li') {
+  if (process.env.NODE_ENV === 'production') {
+    return;
+  }
+
+  if (renderedAs === 'div' && parentRenderedAs !== 'div') {
+    throw new Error('ListItem cannot be rendered as a div when its parent is not a div.');
+  }
+  if (renderedAs === 'li' && parentRenderedAs === 'div') {
+    throw new Error('ListItem cannot be rendered as a li when its parent is a div.');
+  }
+}
+
+function validateNavigableWhenOnClickPresent(navigable: boolean, onClick?: React.MouseEventHandler) {
+  if (process.env.NODE_ENV === 'production') {
+    return;
+  }
+
+  if (onClick && !navigable) {
+    throw new Error('ListItem must be navigable when onClick is present. Set navigable={true} on the List.');
+  }
+}
+
+/**
+ * Create the state required to render ListItem.
+ *
+ * The returned state can be modified with hooks such as useListItemStyles_unstable,
+ * before being passed to renderListItem_unstable.
+ *
+ * @param props - props from this instance of ListItem
+ * @param ref - reference to root HTMLLIElement | HTMLDivElementof ListItem
+ */
+export const useListItem_unstable = (
+  props: ListItemProps,
+  ref: React.Ref<HTMLLIElement | HTMLDivElement>,
+): ListItemState => {
+  const id = useId('listItem');
+  const { value = id, truncateHeader, truncateContent, onClick, onKeyDown, onKeyUp } = props;
+
+  const navigable = useListContext_unstable(ctx => ctx.navigable);
+  const toggleItem = useListContext_unstable(ctx => ctx.selection?.toggleItem);
+  const isSelectionEnabled = useListContext_unstable(ctx => !!ctx.selection);
+  const isSelected = useListContext_unstable(ctx => ctx.selection?.isSelected(value));
+
+  const truncateHeaderOnList = useListContext_unstable(ctx => ctx.truncateHeader);
+  const truncateContentOnList = useListContext_unstable(ctx => ctx.truncateContent);
+
+  const parentRenderedAs = useListContext_unstable(ctx => ctx.as);
+  const renderedAs = props.as || DEFAULT_ROOT_EL_TYPE;
+
+  validateProperElementTypes(parentRenderedAs, renderedAs);
+
+  validateNavigableWhenOnClickPresent(navigable, onClick);
+
+  const focusableGroupAttrs = useFocusableGroup({ tabBehavior: 'limited-trap-focus' });
```

Comment:
```suggestion
```

This shouldn't be needed in this context, since v0 never supported any nested focusables in the list

Reply:
fair point, it can always be added by the consumer later if needed.